Automattic/woocommerce-payments
# Fix WooPay express checkout behavior with product bundles on product page

Fixes #7157
Fixes #6634

#### Changes proposed in this Pull Request








* Remove the WooPay express checkout button disabled state and show an alert when it's clicked with the `Add to cart` button disabled.
* Update the WooPay `add_to_cart` hook to support product bundles, this way the bundle will be added to the cart with the expected configuration before the redirect.

#### Testing instructions





* Create a variable product.
* Access the product page.
* Click WooPay button without select any options.
* An alert should be shown.
* Fill out the variable product options.
* Click `Clear` button or reset an option.
* Click WooPay button.
* An alert should be shown.
* Fill out the variable product options.
* Click the WooPay button, you should be redirected as expected.
* Create a bundle with and without variable products.
* Access the bundle without the variable products.
* Click the WooPay button, you should be redirected as expected and no empty cart error message should be shown.
* Access the bundle with the variable products.
* Click WooPay button without select any options.
* An alert should be shown.
* Fill out the variable product options.
* Click `Clear` button or reset an option on any variable product.
* Click WooPay button.
* An alert should be shown.
* Fill out the variable product options.
* The WooPay button should be enabled.
* Click the WooPay button, you should be redirected as expected and no empty cart error message should be shown.

-------------------


**Post merge**
